### PR TITLE
chore: remove redundant '@JvmStatic' annotations

### PR DESCRIPTION
**Pull request by david-allison**
We're no longer in Java, so only a few annotations are still necessary.

These are mostly test methods (`@Parameters/@MethodSource`), a few which are useful for mocks, and a small number which cause tests to fail if they're removed for no reasonable reason

This is a potentially flaky commit, hopefully the reduction in code causes a reduction in compile times which makes this worthwhile

This was a mostly automated process, adding comments manually if removing the annotation failed tests

It did not modify the API project public classes

## Checklist
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Review by lukstbit — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/test/java/com/ichi2/anki/servicemodel/UpgradeGesturesToControlsTest.kt [hunk 1]
```diff
@@ -178,7 +178,6 @@ class UpgradeGesturesToControlsTest(private val testData: TestData) : Robolectri
         private val volume_up_binding = MappableBinding(keyCode(KEYCODE_VOLUME_UP), Reviewer(CardSide.BOTH))
         private val volume_down_binding = MappableBinding(keyCode(KEYCODE_VOLUME_DOWN), Reviewer(CardSide.BOTH))
 
-        @JvmStatic
```
[the remark this answers is not in the document]
Comment (david-allison) [reply AnkiDroid/src/test/java/com/ichi2/anki/servicemodel/UpgradeGesturesToControlsTest.kt:181]: Test is `Ignored`, great catch!

**Review by lukstbit — APPROVED** [review]
LGTM

**Review by mikehardy — COMMENTED** [review]
```
AnkiStatsTaskHandlerTest > testCreateReviewSummaryStatistics FAILED
    kotlinx.coroutines.test.UncompletedCoroutinesError: After waiting for 60000 ms, the test coroutine is not completing
        at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt$runTestCoroutine$3$3.invokeSuspend(TestBuilders.kt:342)
        (Coroutine boundary)
        at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt.runTestCoroutine(TestBuilders.kt:326)
        at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt$runTest$1$1.invokeSuspend(TestBuilders.kt:167)
        at kotlinx.coroutines.test.TestBuildersJvmKt$createTestResult$1.invokeSuspend(TestBuildersJvm.kt:13)

        Caused by:
        kotlinx.coroutines.test.UncompletedCoroutinesError: After waiting for 60000 ms, the test coroutine is not completing
            at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt$runTestCoroutine$3$3.invokeSuspend(TestBuilders.kt:342)
            at app//kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
            at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
            at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:284)
            at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
            at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
            at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
            at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
            at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
            at kotlinx.coroutines.test.TestBuildersJvmKt.createTestResult(TestBuildersJvm.kt:12)
            at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt.runTest(TestBuilders.kt:166)
            at kotlinx.coroutines.test.TestBuildersKt.runTest(Unknown Source)
            at kotlinx.coroutines.test.TestBuildersKt__TestBuildersKt.runTest(TestBuilders.kt:154)
            at kotlinx.coroutines.test.TestBuildersKt.runTest(Unknown Source)

```

:thinking:  ?

(locally - target `testPlayDebugUnitTest`)

**Comment by david-allison** [comment]
Mentioned (with a potentially pending fix): https://github.com/ankidroid/Anki-Android/issues/11091#issuecomment-1234694172

**Review by mikehardy — APPROVED** [review]

**Pull request merged**